snuba/query/processors/arrayjoin_keyvalue_optimizer.py

```diff
@@ -57,7 +57,7 @@ def _get_mapping_keys_in_condition(
     """
     keys_found = set()
 
-    conditions = get_first_level_conditions(condition)
+    conditions = get_first_level_and_conditions(condition)
     for c in conditions:
         if is_binary_condition(c, BooleanFunctions.OR):
             return None
```

Comment:
Still relevant? Seems like these are already filtered out?

Reply:
they are not filtered out. So this is still relevant.
If the original conditions is `a AND (b OR c)`
get_first_level_and_condition returns
`['a', 'b OR c']`
So we still want to know if any of those top level condition is an OR. This has to return None since `b OR c` is an OR.